### PR TITLE
optimize: remove druid dependency in ConnectionProxy

### PR DESCRIPTION
**Pull request by l81893521**
- [ ] I have registered the PR [changes](https://github.com/seata/seata/tree/develop/changes).

### Ⅰ. Describe what this PR did
remove the druid dependency in ConnectionProxy

### CODE REVIEW

**Review by a364176773 — APPROVED** [review]
LGTM

File: changes/en-us/develop.md [hunk 1]
```diff
@@ -70,6 +70,7 @@ Add changes here for all PR submitted to the develop branch.
 - [[#5047](https://github.com/seata/seata/pull/5047)] remove useless code
 - [[#5051](https://github.com/seata/seata/pull/5051)] undo log dirty throw BranchRollbackFailed_Unretriable
 - [[#5075](https://github.com/seata/seata/pull/5075)] intercept the InsertOnDuplicateUpdate statement which has no primary key and unique index value
+- [[#5104](https://github.com/seata/seata/pull/5075)] remove the druid dependency in ConnectionProxy
```
Comment (caohdgege) [inline changes/en-us/develop.md:73]: url错了，下同

Comment (l81893521) [reply changes/en-us/develop.md]: fixed

**Review by caohdgege — APPROVED** [review]
LGTM

**Review by wangliang181230 — APPROVED** [review]
LGTM

**Review by slievrly — COMMENTED** [review]
LGTM

**Pull request merged**